pkgs/stdenv/generic/check-meta.nix

```diff
@@ -34,23 +36,29 @@ let
   hasLicense = attrs:
     attrs ? meta.license;
 
-  hasAllowlistedLicense = assert areLicenseListsValid; attrs:
-    hasLicense attrs && lib.lists.any (l: builtins.elem l allowlist) (lib.lists.toList attrs.meta.license);
+  hasListedLicense = assert areLicenseListsValid; list: attrs:
+    length list > 0 && hasLicense attrs && any (l: elem l list) (toList attrs.meta.license);
```

Comment:
Is there a non-ugly way to get rid of that `toList` as well?

Reply:
Done. Though it's worth noting that this branch is only hit when you have either allow/block listed licenses, so I'd wager the impact is tiny.